Settlement menu open twice fix

DiMiGi opened the pull request:
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] All new classes have class-level documentation comments, if there are any at all
- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Other... Please describe:


## What is the current behaviour?
When leaving a settlement the menu opens again, after pressing a second time the "Leave" button the party would leave the settlement


## What is the new behaviour?
Resolves #134 

When leaving a settlement the menu opens once

## Other information

DiMiGi committed:
Menu opens once

DiMiGi committed:
Reorder on FieldChangeBuffer

The changes are made just when there is a new value

araex commented on source/Coop/Mod/Persistence/Party/MovementData.cs:
```diff
@@ -93,6 +93,20 @@ public bool NearlyEquals(MovementData other)
+
+            MovementData other = (MovementData)obj;
+            return DefaultBehaviour.Equals(other.DefaultBehaviour) &&
+                   TargetPosition.Equals(other.TargetPosition) &&
```
Do you really want to exactly compare the float coordinates? I think the position should be compared using `Compare.CoordinatesEqual(Vec2 a, Vec2 b)`. As map coordinates are compressed for synchronization (`Compression.Coordinate2d`), floating point errors are to be expected. I realize you're using the new Equals method to compare local values only, but for the sake of usability i think it would be best to implement it using the coordinate compare anyways. This might prevent comparision errors in the future. Or what do you think?

Looking at the class, there's already a `MovementData.NearlyEquals(MovementData other)` which should use `Compare.CoordinatesEqual(Vec2 a, Vec2 b)` as well. I think it would be good if you combined `Equals` and `NearlyEquals` into a single method, probably just `Equals`.

DiMiGi replied:
You're right, I've changed the implementation but instead of using CoordinatesEqual I've used the NearlyEquals implemented by TaleWorlds in the Vec2 Class, using the precision in the Compare Class.

![image](https://user-images.githubusercontent.com/23215672/103768370-24f29880-5001-11eb-902a-166c56a288bc.png)

araex reviewed (commented):
Looks good, but i'd like to discuss the implementation of `MovementData.Equals`. I think it should compensate for the floating point errors expected due to the synchronization compression. Even if it currently is only used to compare non-synchronized values. But i'm not sure what the best practices are in C# regarding `Equals`. 

Does anyone else have an opinion on this? Otherwise i'll leave it up to @DiMiGi to decide.

DiMiGi committed:
Fix float error precision on Equals

Equals methods is override to use it as object type (without casting to MovementData). NearlyEquals implementation is already done by Taleworlds in a Vec2 type data which considers a precision of 1E-5 by default.

DiMiGi commented:
As far as I know `Equals` is used to compare objects and not to know if these objects are the same instance, that means `Equals` should be used in this case. Also, it brings abstraction when using `System.Object` instances, because casting won't be needed, as you could see in source/Sync/FieldChangeBuffer.cs file, where `Equals` is being used in a `Object` type instance, but instead of using the `Object` type implementation it's actually using the `Equals` of the `MovementData` type.
